fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -545,6 +545,15 @@ private ScoredPlan planIndex(@Nonnull PlanContext planContext, @Nonnull QueryCom
                     p = computeIndexFilters(planContext, p);
                 }
                 return p;
+            } else if (indexTypes.getLuceneTypes().contains(index.getType())) {
+                p = planLucene(candidateScan, index, filter, sort);
+                if (p != null) {
+                    p = planRemoveDuplicates(planContext, p);
+                }
+                if (p != null) {
+                    p = computeIndexFilters(planContext, p);
```

Comment:
Hmm -- I think this can only ever find an index filter for a predicate that is expressed on a part of the primary key of the type. Can you add a test case to check that it's working as expected, i.e.

filter: `x = 5` and `luceneComponent` and `x` is part of the primary key of the type
you should see that the index filters being computed contain `x=5`

Reply:
Handled in line 239 of FDBLuceneQueryTest (https://github.com/FoundationDB/fdb-record-layer/pull/1165/files#diff-9c5e48f9f4371e68d15f5f0dd1755b60c23651214984a6a0ea5508ba14cc8352R239)